ansible_rulebook/rule_set_runner.py

```diff
@@ -284,6 +286,29 @@ def _handle_action_completion(self, task):
             if not self.action_loop_task.done():
                 self.action_loop_task.cancel()
 
+    async def _run_action_with_lock(
+        self,
+        lock_name: str,
+        action: str,
+        metadata: Metadata,
+        control: Control,
+        action_args: dict,
+    ):
+        """
+        Acquire the lock for the specified resource, access the resource,
+        and release the lock.
+        """
+        async with self.locks[
+            lock_name
```

Comment:
The current implementation assumes the lock name is unique ruleset, if multiple actions define the same name it would have unexpected behaviors.  This is not explained in the docs. 

Do we want to support the same lock across multiple actions? I feel we should not, in that case the application should validate it.

Reply:
When I say multiple actions, I mean multiple rules, because I understand the scope of the lock is per rule (a rule certainly could have multiple actions, in that case, would make sense)